src/components/Header/Header.tsx

```diff
@@ -1,11 +1,43 @@
 import AppBar from "@mui/material/AppBar";
 import Toolbar from "@mui/material/Toolbar";
 import Button from "@mui/material/Button";
+import jwt_decode from "jwt-decode";
 import { useNavigate } from "react-router-dom";
-import { Grid } from "@mui/material";
+import { Avatar, Grid } from "@mui/material";
+import { authApi } from "../../services/auth";
+import { useEffect } from "react";
+import { getUserName } from "../../slices/userSlice";
+import { useAppDispatch, useAppSelector } from "../../store/hooks";
+import { AUTH_TOKEN } from "../../constants/constants";
+import { Token } from "../../types";
 
 export const Header = () => {
+  const { name } = useAppSelector((state) => state.user);
+  const dispatch = useAppDispatch();
+
   const navigate = useNavigate();
+
+  const logout = () => {
+    dispatch(getUserName(""));
+    localStorage.setItem(AUTH_TOKEN, "");
+    dispatch(authApi.util.resetApiState());
+  };
+
+  const token = localStorage.getItem(AUTH_TOKEN);
+
+  useEffect(() => {
+    if (token) {
+      const decoded = jwt_decode<Token>(token);
+      const currentTime = Date.now() / 1000;
+
+      if (decoded.exp < currentTime) {
+        logout();
+      } else {
+        dispatch(getUserName(decoded.name));
```

Comment:
Instead of just name would be better to decode name and id from token and set put it to user slice